src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/TriviaDataFactory.CodeShapeAnalyzer.cs

```diff
@@ -106,6 +107,13 @@ private bool UseIndentation
 
             private static bool OnElastic(SyntaxTrivia trivia)
             {
+                // if this is structured trivia then we need to check for elastic trivia in any descendant
+                if (trivia.HasStructure)
+                {
+                    var structure = trivia.GetStructure()!;
```

Comment:
@jcouv is there any annotation taht says "this method returns non-null if this other property was true"?

Reply:
@CyrusNajmabadi I think only if `GetStructure` was a property getter, then `MemberNotNullWhen` would work.


Anyway, since `GetStructure` simply checks for `HasStructure`, and just returns null instead of throwing, I'd change the code to:

```csharp
var structure = trivia.GetStructure();
if (structure is not null)
{
    foreach (var t in structure.DescendantTrivia()) { ... }
}
```

Reply:
since it only checks elastic, I would check whether trivia at least has annotation (elastic is an annotation on trivia) before digging into structured trivia (so, normal formatting such as document format never dig in to find elastic trivia)

and I believe there is a flag on the root of structured trivia whether there is an annotation associated with itself and its children.

https://github.com/dotnet/roslyn/blob/9dc51cfd0b45ff87c595bd35d4edb65b743930f9/src/Compilers/Core/Portable/Syntax/SyntaxNode.cs#L1094

so, something like
if (structure?.ContainsAnnotations) { ... }

Reply:
Yeah, nice.. that lets me do `if (trivia.GetStructure() is { ContainsAnnotations: true } structure)` and combine both suggestions.

Of course putting the code here has broken change signature, so down the rabbit hole I go... tomorrow :)

Reply:
We also run into this with `SyntaxNodeOrToken.IsNode`/`AsNode()`.

Reply:
Hrmm .  That Seems Unfortunate